src/main/kotlin/app/revanced/patcher/extensions/InstructionExtensions.kt

```diff
@@ -11,7 +12,23 @@ import org.jf.dexlib2.builder.MutableMethodImplementation
 import org.jf.dexlib2.builder.instruction.*
 import org.jf.dexlib2.iface.instruction.Instruction
 
+
+class InstructionInserter(val method: MutableMethod, startIndex: Int) {
+    var index = startIndex
+    fun putInstructions(smaliInstructions: String) {
+        putInstructions(smaliInstructions.toInstructions(method))
+    }
+
+    fun putInstructions(instructions: List<BuilderInstruction>) {
+        method.implementation!!.addInstructions(index, instructions)
+        index += instructions.size
+    }
+}
+
 object InstructionExtensions {
+    fun MutableMethod.insertInstructions(startIndex: Int, fn: InstructionInserter.() -> Unit): Int {
+        return InstructionInserter(this, startIndex).apply(fn).index
+    }
```

Comment:
To adhere to consistency, this extension function should be added for the following signatures, just like all the other extension functions:

- `MutableMethod.insertInstructions(startIndex: Int, fn: InstructionInserter.() -> Unit)`
- `MutableMethod.insertInstructions(fn: InstructionInserter.() -> Unit)`
- `MutableMethodImplementation.insertInstructions(startIndex: Int, fn: InstructionInserter.() -> Unit)`
- `MutableMethodImplementation.insertInstructions(fn: InstructionInserter.() -> Unit)`

Reply:
Adding an extension method on `MutableMethodImplementation` won't be possible since in `InstructionInserter` we have a call to `String.toInstructions` which requires a `MutableMethod` instance

I have added a default value for `startIndex` to `MutableMethod.insertInstructions`, is that fine or do you prefer a second method?

Reply:
> I have added a default value for startIndex to MutableMethod.insertInstructions, is that fine or would you prefer a second method?

No other extension method in this class uses a default value out of the reason that you would have to specify the second parameter and out of the reason because `List.add` doesn't do that either and introduces a second overload.

Reply:
Alright, done.